inloop/solutions/prettyprint/tools.py

```diff
@@ -0,0 +1,303 @@
+"""
+Tools and infrastructure to convert XML to template-friendly dicts.
+
+Numerous tools such as Ant, Maven and Gradle produce test reports
+according to this format, which is described as XML DTD in the
+Javadoc for the class
+
+    org.apache.tools.ant.taskdefs.optional.junit.XMLConstants
+
+which is part of Ant (see https://github.com/apache/ant).
+"""
+from defusedxml import ElementTree as ET
+
+
+class XMLContextParser(object):
+    """
+    Manages the context parsing between Docker XML files and templates.
+
+    When created, a solution is parsed, which contains the xml we need to
+    transform and parse our data in directly processable dicts.
+    """
+
+    def __init__(self, *, solution):
+        """
+        Initializes this object. Unwraps the testoutput_set of the most recent test result
+        to the internal variable self.testoutput.set.
+
+        Args:
+            solution (QuerySet): solution containing XML files.
+
+        """
+        if solution:
+            result = solution.testresult_set.last()
+            self.testoutput_set = result.testoutput_set
+        else:
+            raise ValueError("Solution must not be empty!")
+
+    def __str__(self):
+        """
+        Describes this object
+
+        Returns:
+            str: A string describing this object
+        """
+        return "<XMLParser object with testoutput set: {}>".format(self.testoutput_set)
```

Comment:
This should be returned by `__repr__()`, and the `__str__()` method should be removed, because the default implementation of the built-in `object` returns `obj.__repr__()` (https://docs.python.org/3.7/reference/datamodel.html#object.__str__).

Reply:
I fixed this in #db0111c